test/building.jl

```diff
@@ -273,3 +273,39 @@ end
     end
 end
 
+@testset "Building from remote file" begin
```

Comment:
I've added a very simple test for building a remote source, as this code path is completely untested.  If you have better ideas about what and how to test here, please share

Reply:
This seems fine to me; it does mean that our tests may get sad if the network goes down; we could run a local webserver to fix that, but let's cross that bridge once we get to it.